microdnf-plugins/changelog_plugin/changelog.cpp

```diff
@@ -0,0 +1,228 @@
+*/
+
+
+#include "changelog.hpp"
+
+#include "microdnf/context.hpp"
+
+#include <libdnf/conf/option_string.hpp>
+#include <libdnf/rpm/package.hpp>
+#include <libdnf/rpm/package_query.hpp>
+#include <libdnf/rpm/package_set.hpp>
+
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+
+namespace microdnf {
+
+
+using namespace libdnf::cli;
+
+
+ChangelogCommand::ChangelogCommand(Command & parent) : Command(parent, "changelog") {
+    auto & ctx = static_cast<Context &>(get_session());
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("Show changelog data of packages");
+
+    since_option = dynamic_cast<libdnf::OptionNumber<std::int64_t> *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionNumber<std::int64_t>>(
+            new libdnf::OptionNumber<int64_t>(0, [](const std::string & value) {
+                struct tm time_m;
+                std::istringstream ss(value);
+                ss >> std::get_time(&time_m, "%Y-%m-%d");
+                if (ss.fail()) {
+                    throw std::runtime_error(fmt::format("Invalid date: {}", value));
+                }
+                return static_cast<int64_t>(mktime(&time_m));
+            }))));
+
+    count_option = dynamic_cast<libdnf::OptionNumber<std::int32_t> *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionNumber<std::int32_t>>(new libdnf::OptionNumber<int>(0))));
+
+    upgrades_option = dynamic_cast<libdnf::OptionBool *>(
+        parser.add_init_value(std::unique_ptr<libdnf::OptionBool>(new libdnf::OptionBool(false))));
+
+    auto since = parser.add_new_named_arg("since");
+    since->set_long_name("since");
+    since->set_short_description(
+        "show changelog entries since DATE. To avoid ambiguosity, YYYY-MM-DD format is required.");
+    since->set_has_value(true);
+    since->link_value(since_option);
+
+    auto count = parser.add_new_named_arg("count");
+    count->set_long_name("count");
+    count->set_short_description("show given number of changelog entries per package");
+    count->set_has_value(true);
+    count->link_value(count_option);
+
+    auto upgrades = parser.add_new_named_arg("upgrades");
+    upgrades->set_long_name("upgrades");
+    upgrades->set_short_description(
+        "show only new changelog entries for packages, that provide an upgrade for some of already installed "
+        "packages.");
+    upgrades->set_const_value("true");
+    upgrades->link_value(upgrades_option);
+
+    pkgs_spec_to_show_options = parser.add_new_values();
+    auto keys = parser.add_new_positional_arg(
+        "pkg_spec",
+        ArgumentParser::PositionalArg::UNLIMITED,
+        parser.add_init_value(std::unique_ptr<libdnf::Option>(new libdnf::OptionString(nullptr))),
+        pkgs_spec_to_show_options);
+    keys->set_short_description("List of packages specifications");
```

Comment:
> list of package specifiers

(note no capital L)

I think "specifiers" is a better form to use here but I actually don't know how wide-spread is the word "specification" for this.

Reply:
It is a description of a positional argument. We start with a capital letter in all commands. We should discuss these descriptions and then possibly make a change in all commands to keep the descriptions consistent.